src/battle/battle_input.c

```diff
@@ -0,0 +1,4519 @@
+    MoveTypeIcon_LoadPalette(BattleSystem_GetPaletteData(battleInput->battleSystem), 3, spriteSystem, spriteManager, NNS_G2D_VRAM_TYPE_2DSUB, 20020);
+
+    for (i = 0; i < 4; i++) {
+        sub_0207C948(spriteSystem, spriteManager, NNS_G2D_VRAM_TYPE_2DSUB, TYPE_NORMAL, 20025 + i);
+    }
+
+    if (BattleSystem_GetBattleType(battleInput->battleSystem) & BATTLE_TYPE_CATCH_TUTORIAL) {
+        CatchingTutorialFinger_LoadResources(spriteSystem, spriteManager, HEAP_ID_BATTLE, palette, 20412, 20036, 20406, 20395);
+        battleInput->catchingTutorial.finger = CatchingTutorialFinger_Create(spriteSystem, spriteManager, HEAP_ID_BATTLE, 20412, 20036, 20406, 20395, 10, 0);
+    }
+}
+
+void ov16_02268B8C(BattleInput *battleInput)
+{
+    int i;
+    SpriteSystem *spriteSystem = BattleSystem_GetSpriteSystem(battleInput->battleSystem);
+    SpriteManager *spriteManager = BattleSystem_GetSpriteManager(battleInput->battleSystem);
+
+    for (i = 0; i < 4; i++) {
+        sub_0207CA34(spriteManager, 20025 + i);
+    }
+
+    sub_0207CA3C(spriteManager, 20020);
+    sub_0207CA44(spriteManager, 20017, 20017);
+
+    if (BattleSystem_GetBattleType(battleInput->battleSystem) & BATTLE_TYPE_CATCH_TUTORIAL) {
+        CatchingTutorialFinger_Free(battleInput->catchingTutorial.finger);
+        CatchingTutorialFinger_FreeResources(spriteManager, 20412, 20036, 20406, 20395);
+    }
+}
+
+void BattleInput_ChangeMenu(NARC *narcUnused, NARC *param1, BattleInput *battleInput, int menuId, int param4, void *param5)
+{
+    const BattleMenuTemplate *menuTemplate, *prevMenuTemplate;
+    int i;
+    BgConfig *bgConfig;
+    SpriteSystem *spriteSystem;
+    SpriteManager *spriteManager;
+
+    if (param5 != NULL) {
+        MI_CpuCopy8(param5, &battleInput->menu, sizeof(BattleInputMenu));
+    }
+
+    battleInput->touchDisabled = FALSE;
+
+    bgConfig = BattleSystem_GetBgConfig(battleInput->battleSystem);
+    spriteSystem = BattleSystem_GetSpriteSystem(battleInput->battleSystem);
+    spriteManager = BattleSystem_GetSpriteManager(battleInput->battleSystem);
+
+    if (battleInput->curMenuId == -1) {
+        param4 = 1;
+        prevMenuTemplate = NULL;
+    } else {
+        prevMenuTemplate = &sBattleMenuTemplates[battleInput->curMenuId];
+    }
+
+    menuTemplate = &sBattleMenuTemplates[menuId];
+
+    PaletteData_LoadBuffer(BattleSystem_GetPaletteData(battleInput->battleSystem), battleInput->unk_58, 1, 0, 0x200);
+
+    for (i = 0; i < 4; i++) {
+        if ((menuTemplate->unk_04_val2[i] != 0xffff) && ((param4 == 1) || (menuTemplate->unk_04_val2[i] != prevMenuTemplate->unk_04_val2[i]))) {
+            Bg_LoadTilemapBuffer(bgConfig, 4 + i, battleInput->unk_3C[menuTemplate->unk_04_val2[i]], 0x800);
+            Bg_ScheduleTilemapTransfer(bgConfig, 4 + i);
+        }
+    }
+
+    SpriteSystem_LoadPaletteBufferFromOpenNarc(BattleSystem_GetPaletteData(battleInput->battleSystem), 3, spriteSystem, spriteManager, param1, 72, 0, 7, NNS_G2D_VRAM_TYPE_2DSUB, 20023);
```

Comment:
```suggestion
    SpriteSystem_LoadPaletteBufferFromOpenNarc(BattleSystem_GetPaletteData(battleInput->battleSystem), PLTTBUF_SUB_OBJ, spriteSystem, spriteManager, param1, 72, FALSE, 7, NNS_G2D_VRAM_TYPE_2DSUB, 20023);
```

Applies to decorating other instances in documented functions.